src/test/java/in/projecteka/consentmanager/link/discovery/DiscoveryUserJourneyTest.java

```diff
@@ -164,4 +193,118 @@ public void initialize(ConfigurableApplicationContext applicationContext) {
             values.applyTo(applicationContext);
         }
     }
+
+    @Test
+    public void shouldGetGatewayTimeoutForDiscoverCareContext() throws Exception {
+        var token = string();
+        String requestId = "cecd3ed2-a7ea-406e-90f2-b51aa78741b9";
+        String patientDiscoveryRequest = "{\n" +
+                "  \"requestId\": \""+ requestId + "\",\n" +
+                "  \"hip\": {\n" +
+                "    \"id\": \"12345\"\n" +
+                "  }\n" +
+                "}";
+        String userId = "test-user-id";
+        when(authenticator.verify(token)).thenReturn(Mono.just(new Caller(userId, false)));
+        when(userServiceClient.userOf(userId)).thenReturn(Mono.just(TestBuilders.user().build()));
+        when(discoveryRepository.getIfPresent(any())).thenReturn(Mono.empty());
+        when(discoveryRepository.insert(anyString(), anyString(), any(), any())).thenReturn(Mono.empty());
+        when(discoveryServiceClient.requestPatientFor(any(), eq("http://tmc.gov.in/ncg-gateway"), eq("12345"))).thenReturn(Mono.just(PatientResponse.builder().build()));
```

Comment:
Shouldn't this be a test to introduce intentional delay beyond the annotated timeout?

Reply:
Think its ok for now. ideally yes.